### DISCUSSION TITLE
how to deserialize from json

### DISCUSSION THREAD

**Discussion by hans000** [facebook/lexical#2578] [Q&A] [answered]
how to restore by json. it cannot work.
```
const snapshotList = []

// create a snapshot 
const editorState = editor.getEditorState()
const snapshot = editorState.toJSON()
snapshotList.push(snapshot)

// restore
editor.update(() => {
   const root = $getRoot()
   const node = RootNode.importJSON(snapshotList[<index>])

  root.append(...node.getChildren())
})


```

**Comment by hans000** [comment, marked as answer]
i find it

```
editor.update(() => {
  const editorState = editor.parseEditorState(data)
  editor.setEditorState(editorState);
})
```

**Reply by sneko** [reply]
@hans000 do you know how to do it from the `LexicalComposer`?

When giving it a try through `initialConfig.editorState` I get:
```
Cannot read properties of undefined (reading 'type')
```

Whereas it's supposed to accept JSON stringified apparently:
```
type InitialEditorStateType = string | EditorState | ((editor: LexicalEditor) => void) | null
```

**EDIT:**
... ok found the issue! My stringified content came from the Playground with "export" function... which wraps everything into a `{editorState:{...}}`. It seems it's not the standard way for the `LexicalComposer` to be initialized, so after removing this wrapper to directly get `{root:{...}}` it works :)

**Reply by YuluDuan** [reply]
hello, I have the same problem here How do you remove the wrapper `{editorState:{...}}` thanks!
